### ISSUE TITLE
Change of default value `axis=-1` for BatchNormalization

### ISSUE THREAD

**Issue by dolaameng** [fchollet/keras#3888]
I am referring to `axis=-1` in the [BatchNormalization.__init__()](https://github.com/fchollet/keras/blob/master/keras/layers/normalization.py#L63). Would it make more sense to set its default values based on `image_dim_ordering`? 

For example, with 'th' dim_ordering, the following code needs to explicitly set the `axis=1` (otherwise an exception will be raised), which could be a burden on the user side?
```python
x = Input(shape=(1, 28, 28))
y = Conv2D(128, 3, 3, activation='relu', border_mode='same')(x)
y = BatchNormalization(mode=2, axis=1)(y)
y = Flatten()(y)
y = Dense(6, activation='softmax')(y)
model = Model(input=x, output=y)
model.compile(loss='categorical_crossentropy', optimizer='adam', metrics=['accuracy'])
```

A discussion can be found [here](https://github.com/osh/KerasGAN/issues/2)

**Comment by dolaameng** [comment]
Looks like the idea is not popular. Examples of code can be found in [here](https://github.com/fchollet/keras/blob/master/keras/applications/resnet50.py#L43) anyway.

**Comment by Neltherion** [comment]
after updating to CUDA 8 and the latest Keras+Theano on windows, I've been having the same problem. I think the idea is not popular because not everyone has updated their Keras yet but it's crucial to be addressed.
